flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -742,6 +755,24 @@ public FileVisitResult preVisitDirectory(java.nio.file.Path dir, BasicFileAttrib
 			}
 		}
 
+		for (URI originURI : archives) {
+			Path remoteParent = Utils.getRemoteResourceRoot(appId.toString(), new Path(originURI.getPath()), fs.getHomeDirectory());
+			String fragment = originURI.getFragment();
+			Path target = new Path(
+				FileUtils.localizeRemoteFiles(new org.apache.flink.core.fs.Path(remoteParent.toUri()), originURI).toUri());
+			URI targetURI = target.toUri();
+			if (targetURI.getFragment() == null && fragment != null) {
+				targetURI = new URI(target.toUri().toString() + "#" + fragment);
```

Comment:
Why do we need to append a fragment here? I thought if `remoteParent` contained a fragment, then `target` would be a file of that fragment and be named after the fragment.